### PR TITLE
LPD-27358 Allow scheduled/expired versions to be deleted/reverted to

### PR DESCRIPTION
**Pull request by manuelRives**
# What is this trying to solve?

- https://liferay.atlassian.net/browse/LPD-27358 Allow scheduled/expired versions to be deleted/reverted to

# How am I fixing it?

Modifying the revert and delete methods so that they do not throw an exception if the status is scheduled or expired

# How can you verify that it works?

a. Scheduled content
  1. Create a scheduled content. 
  2. Modify it to generate a new version.
  3. Delete this new version from the interface.

b. Expired content
  1. Create a content. 
  2. Modify it to generate a new version and assign the current time as expiration date.
  3. Wait for it to expire.
  4. Delete this new version from the interface.

### CODE REVIEW

**Comment by manuelRives** [comment]
ci:test:sf

File: modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/service/test/DLFileEntryLocalServiceTest.java [hunk 1]
```diff
@@ -798,6 +798,29 @@ public void testCopyFileEntry() throws Exception {
...
+
+		dlFileEntry = updateDLFileEntryWithStatus(
+			dlFileEntry, new ByteArrayInputStream(new byte[0]), new HashMap<>(),
+			serviceContext, WorkflowConstants.STATUS_EXPIRED);
+
+		DLFileEntryLocalServiceUtil.deleteFileVersion(
+			TestPropsValues.getUserId(), dlFileEntry.getFileEntryId(),
+			dlFileEntry.getLatestFileVersion(
+				true
+			).getVersion());
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/service/test/DLFileEntryLocalServiceTest.java:818]: This kind of method chaining is not allowed.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/service/test/DLFileEntryLocalServiceTest.java (hunk 1 above)
Comment (manuelRives) [reply modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/service/test/DLFileEntryLocalServiceTest.java]: I completely missed it, thank you very much for the heads up.
Fixed.

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java [hunk 2]
```diff
@@ -885,17 +885,29 @@ public DLFileEntry deleteFileVersion(
...
-			int count = _dlFileVersionPersistence.countByF_S(
+			int fileVersionsCount = _dlFileVersionPersistence.countByF_S(
 				fileEntryId, WorkflowConstants.STATUS_APPROVED);
 
-			if (count <= 1) {
+			fileVersionsCount += _dlFileVersionPersistence.countByF_S(
+				fileEntryId, WorkflowConstants.STATUS_SCHEDULED);
+
+			int fileVersionsExpired = _dlFileVersionPersistence.countByF_S(
+				fileEntryId, WorkflowConstants.STATUS_EXPIRED);
```
[the remark this answers is not in the document]
Comment (manuelRives) [reply portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java]: Fixed.

File: modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/service/test/DLFileEntryLocalServiceTest.java [hunk 3]
```diff
@@ -1195,6 +1241,32 @@ public void testMoveFileEntryToInvalidDLFolder() throws Exception {
...
+
+		DLFileEntryLocalServiceUtil.revertFileEntry(
+			TestPropsValues.getUserId(), dlFileEntry.getFileEntryId(),
+			originalVersion.getVersion(), serviceContext);
+
+		Assert.assertEquals(
+			originalVersion.getTitle(),
+			dlFileEntry.getLatestFileVersion(
+				true
+			).getTitle());
```
[the remark this answers is not in the document]
Comment (manuelRives) [reply modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/service/test/DLFileEntryLocalServiceTest.java]: I thought I had changed it when you told me about the chaining in the previous review, my bad.

File: modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/service/test/DLFileEntryLocalServiceTest.java [hunk 4]
```diff
@@ -1508,6 +1580,29 @@ protected DLFileEntry addAndApproveFileEntry(
...
+			StringUtil.randomString(), ContentTypes.TEXT_PLAIN,
+			StringUtil.randomString(), StringUtil.randomString(),
+			StringPool.BLANK, StringPool.BLANK, -1, new HashMap<>(), null,
+			new ByteArrayInputStream(new byte[0]), 0, displayDate,
+			expirationDate, new Date(), serviceContext);
+
+		return DLFileEntryLocalServiceUtil.updateStatus(
+			TestPropsValues.getUserId(),
+			dlFileEntry.getFileVersion(
+			).getFileVersionId(),
```
[the remark this answers is not in the document]
Comment (manuelRives) [reply modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/service/test/DLFileEntryLocalServiceTest.java]: Fixed

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:report:15718589

**Comment by robertoDiaz** [comment]
ci:forward

**Comment by robertoDiaz** [comment]
ci:report:19004995

**Comment by robertoDiaz** [comment]
ci:test:relevant

**Comment by robertoDiaz** [comment]
ci:forward

**Comment by robertoDiaz** [comment]
ci:forward:force

**Pull request closed without merging**